### DISCUSSION TITLE
Compatibility of autonomousSequence.py with Lighthouse Basestations

### DISCUSSION THREAD

**Discussion by dakeprithvi** [bitcraze/discussions#1702] [Q&A] [answered]
Hi!

We recently transitioned from the Loco-positioning system to the Lighthouse basestations for our Crazyflie setup. Previously, we used the autonomousSequence.py script from the Crazyflie Python library ([link to script](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/autonomy/autonomousSequence.py)) to feed pre-calculated trajectories. However, the script's documentation mentions that it is designed to work with the GPS system.

Here are my questions:

Can this script work with the Lighthouse basestations? If not, what modifications or alternative approaches would you recommend to make it compatible with the Lighthouse system?

Any guidance or pointers would be greatly appreciated! Thanks in advance for your help!

**Comment by gemenerik** [comment, marked as answer]
This example, although originally developed for the Loco positioning system, should work perfectly with the Lighthouse system as well since there's no Loco-specific code involved. The only function that really stands out is the variance check in the Kalman filter, but this is also applicable to the Lighthouse system. This check is valuable because it ensures the filter has converged. In short, it should work seamlessly out of the box.
